geoengine/auth.py

```diff
@@ -35,14 +36,26 @@ class Session:
 
     session: ClassVar[req.Session] = None
 
-    def __init__(self, server_url: str) -> None:
+    def __init__(self, server_url: str, credentials: Tuple[str, str] = None) -> None:
```

Comment:
i think we should have a credentials (or client) config struct instead of a tuple. Then we can also serialize it and store it as ./.geoengine/client.json or as ~/.config/geoengine/client.json.